### PR TITLE
tweaks to event card header for mobile

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request enhances the `EventCardHeader` component to better support responsive design and accessibility, and adds a comprehensive test suite to ensure correct behavior across device types and user interactions.

**Component improvements:**

* The `EventCardHeader` now uses the `useMuiState` hook to determine if the view is mobile, allowing the header text size to adjust dynamically (`variant` is `'h6'` for mobile, `'h5'` otherwise) and adding right padding for improved layout. (`[[1]](diffhunk://#diff-34cc3aec98ef68eff413d94c6a6d8cb333a43331a204a821cd9b6db9de357876R12)`, `[[2]](diffhunk://#diff-34cc3aec98ef68eff413d94c6a6d8cb333a43331a204a821cd9b6db9de357876R24)`, `[[3]](diffhunk://#diff-34cc3aec98ef68eff413d94c6a6d8cb333a43331a204a821cd9b6db9de357876L35-R39)`)
* The event options button now explicitly sets its size to `"small"` for better alignment and appearance. (`[apps/web/src/components/EventCards/EventCardHeader.tsxR55](diffhunk://#diff-34cc3aec98ef68eff413d94c6a6d8cb333a43331a204a821cd9b6db9de357876R55)`)

**Testing improvements:**

* A new test file, `EventCardHeader.test.js`, was added with thorough tests covering desktop and mobile rendering, dropdown interactions, callback invocations, and edge cases with event names. The `useMuiState` hook is mocked to simulate different device contexts. (`[apps/web/src/components/EventCards/__tests__/EventCardHeader.test.jsR1-R142](diffhunk://#diff-21a3d2a1a654348aceb65bffc30f5622eb1cb888e90ec7c5d53e87dacbcdb64bR1-R142)`)

image

image

### CODE REVIEW

File: apps/web/src/components/EventCards/__tests__/EventCardHeader.test.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    it('hides dropdown when clicking away', async () => {
+        render(<EventCardHeader {...defaultProps} />);
+
+        const optionsButton = screen.getByRole('button', { name: 'Event options for Test Event' });
+        await user.click(optionsButton);
+
+        expect(screen.getByRole('menu')).toBeInTheDocument();
+
+        // Click outside the dropdown
+        user.click(document.body);
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply apps/web/src/components/EventCards/__tests__/EventCardHeader.test.js:78]: not calling await since we're awaiting for an element to be removed right after

File: apps/web/src/components/EventCards/__tests__/EventCardHeader.test.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        render(<EventCardHeader {...defaultProps} />);
+
+        const optionsButton = screen.getByRole('button', { name: 'Event options for Test Event' });
+
+        // Open dropdown
+        await user.click(optionsButton);
+        expect(screen.getByRole('menu')).toBeInTheDocument();
+
+        // Close by clicking outside
+        user.click(document.body);
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply apps/web/src/components/EventCards/__tests__/EventCardHeader.test.js:130]: not calling await since we're awaiting for an element to be removed right after
